src/PlaneWaveBasis.jl

```diff
@@ -332,32 +319,77 @@ end
 
 
 """
-Return the list of wave vectors (integer coordinates) for the cubic basis set.
+    G_vectors(fft_size::Tuple)
+
+The wave vectors `G` in reduced (integer) coordinates for a cubic basis set
+of given sizes.
 """
-function G_vectors(fft_size)
+function G_vectors(fft_size::Tuple)
     start = .- cld.(fft_size .- 1, 2)
     stop  = fld.(fft_size .- 1, 2)
     axes = [[collect(0:stop[i]); collect(start[i]:-1)] for i in 1:3]
     (Vec3{Int}(i, j, k) for i in axes[1], j in axes[2], k in axes[3])
 end
+
+@doc raw"""
+    G_vectors(basis::PlaneWaveBasis)
+    G_vectors(kpt::Kpoint)
+
+The list of wave vectors ``G`` in reduced (integer) coordinates of a `basis`
+or a ``k``-point `kpt`.
+"""
 G_vectors(basis::PlaneWaveBasis) = G_vectors(basis.fft_size)
+G_vectors(kpt::Kpoint) = kpt.G_vectors
 
+
+@doc raw"""
+    G_vectors_cart(basis::PlaneWaveBasis)
+    G_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+
+The list of ``G`` vectors of a given `basis` or `kpt`, in cartesian coordinates.
+"""
 function G_vectors_cart(basis::PlaneWaveBasis)
     (basis.model.recip_lattice * G for G in G_vectors(basis.fft_size))
 end
+function G_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+    (basis.model.recip_lattice * G for G in G_vectors(kpt))
+end
+
+@doc raw"""
+    q_vectors(kpt::Kpoint)
 
+The list of ``q = G + k`` vectors, in reduced coordinates.
 """
-Return the list of r vectors, in reduced coordinates. By convention, this is in [0,1)^3.
+q_vectors(kpt::Kpoint) = ((G + kpt.coordinate) for G in G_vectors(kpt))
+
+@doc raw"""
+    q_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+
+The list of ``q = G + k`` vectors, in cartesian coordinates.
+"""
+function q_vectors_cart(basis::PlaneWaveBasis, kpt::Kpoint)
+    (basis.model.recip_lattice * (G + kpt.coordinate) for G in G_vectors(kpt))
+end
```

Comment:
Hm, this is a great idea but maybe too clever? The trouble is that q_vectors could mean the set of all k+G, for all k and G

Reply:
other issue is that q vectors is not such a standard thing. In fact, q is standard for the *phonon* Brillouin zone, which we'll want to have for phonons... So, kpG? Kinda ugly but descriptive...

Reply:
Yes I agree. How about this fancy `k+G`? Might be too fancy?

Reply:
Hm, yeah, probably too confusing, it's not clear a priori how to write this using tab completion. I think last time we landed on "unicode craziness OK for local variables in functions, but tricky in function names"... kpG, k_plus_G? k_plus_G_vectors is not too bad

Reply:
I went with `kplusG_vectors` (like in `solve_ΩplusK`).

Reply:
OK that makes sense!